### ISSUE TITLE
Bulk delete leaves messages in the view till view refreshes.

### ISSUE THREAD

**Issue by bashok001** [nylas/N1#1523]
After selecting multiple unread emails and then clicking delete, the emails stay on the screen as if nothing happened and deselected. If and when I toggle between different folders and inbox, these emails disappear.

I am facing this issue from a long time. I haven't reported it because I can't reliably capture a screenshot without exposing my emails. Tried hard this time. Here is the gif of the **aftermath of bulk delete**. **Note: This is a full minute or two later delete is executed.** As you see **the emails I deleted** are still on the screen but when I go to another folder and come back, they disappear. Looks like an issue with view recycling.

![try](https://cloud.githubusercontent.com/assets/2344626/13434010/607b944e-dfa2-11e5-9ff7-7f025a4417a2.gif)

- [x] Are there any related issues?
Didn't find any
- [x] What operating system are you using?
Arch Linux
- [x] What version of N1 are you using?
0.4.5 b0ec335

**Bug?**
- [x] Do you have any third-party plugins installed?
Yes, one theme. https://github.com/edipox/n1-ido
- [x] Is the issue related to a specific email provider (Gmail, Exchange, etc.)?
No.
- [x] Is the issue reproducible with a particular attachment, message, signature, etc?
Attached above

**Comment by mbilker** [comment]
0.4.10 is out with a fix for this, though the messages will not disappear until the API request goes through successful. You can complain that there should be an indication the messages are pending remote deletion.

**Comment by bengotow** [comment]
Hey @mbilker this shouldn't be the case. N1 optimistically updates the UI before the API requests complete. I think this is an issue with the thread list query subscription. We've been seeing reports of this assertion failure in Sentry:

```
query-subscription at MutableQuerySubscription.QuerySubscription._createResultAndTrigger
Error: QuerySubscription: Applied all changes and result set contains duplicate IDs.
```

And I think that'd cause the view to not update after an action. We should get to the bottom of this soon!

**Comment by bashok001** [comment]
@mbilker @bengotow I got the update for 0.4.10 and this bug is very much still present for bullk deletes. Just FYI.

**Comment by jstejada** [comment]
@bashok001 , is there any chance you are getting the error @bengotow mentioned in the console?

**Comment by bashok001** [comment]
@jstejada When mails pile up next time and I do a bulk delete, I will surely get that information for you. Stay tuned!

**Comment by mbilker** [comment]
From my experience, N1 wants to do optimistic rendering, but I found that it usually wants the API request to complete. This annoys me while on a cellular hotspot connection.

**Comment by jstejada** [comment]
thanks!

**Comment by mbilker** [comment]
@jstejada I made an edit to my previous post

**Comment by jstejada** [comment]
hmm @mbilker, that's really weird. If you take a look at our tasks, e.g. ChangeMailTask, `performLocal` immediately saves the changes to the database, which in turn triggers and re renders the UI. When `performRemote` completes, it will revert those changes if the request fails.

So we are definitely not waiting for the API request to complete to apply the changes, so it is most likely an issue with the database triggering when there are many changes queued.

Are there any other specific scenarios you've encountered to replicate this issue?

**Comment by jstejada** [comment]
also, this is probably the same issue as #1385

**Comment by mbilker** [comment]
@jstejada I have not been able to reproduce this issue so far in any other cases. I do note it does apply the changes most of the time, but I will have upwards of 8 tasks waiting translating to 8 API requests pending.

**Comment by bashok001** [comment]
Just happened again and I didnt see any errors in my console. I tried deleting again and again as the messages stayed in inbox. Each time the tasks reported success in my dev console. All of these are the tasks working on deleting the same set of messages.

![jjj](https://cloud.githubusercontent.com/assets/2344626/13799229/e8d7ab1e-eaf3-11e5-83e4-254ad99fe004.png)

**Comment by bashok001** [comment]
![try4](https://cloud.githubusercontent.com/assets/2344626/14059674/796287f2-f31f-11e5-9416-3d62ded1e85c.gif)

Bug present in newest version (4.16) too. No errors of any kind seen in activity window.

**Comment by bengotow** [comment]
I think this is a dup of the issue fixed by 765d9f30fc5be75b97e3254d37a6b57fbc341582

**Comment by bengotow** [comment]
Hey folks—this should be fixed in `0.4.32`, which shipped yesterday. Cheers!
